backend/routes/base/branches.js

```diff
@@ -13,6 +13,9 @@ var buildDynamic = function(db, router, auth, revisionService, cache){
 
     const addBranch = async function(repoId, type, name, description, upload_id) {
         const repoBranchSchema = new db.RepoBranchSchema;
+        if (typeof(repoId) === "undefined"){
```

Comment:
Could this be moved to the top for checking passed arguments?

Reply:
Change coming